### PR TITLE
RFC 8484 (binary) support

### PR DESCRIPTION
**Pull request by soumya92**
This PR removes JSON support completely and replaces it with RFC 8484.

From my limited reading, it seems that the JSON protocol is all-but-deprecated, and anyone who supported json also supports the new binary protocol, so this should be a strict improvement.

This should allow more kinds of dns queries, and simpler translation.

### CODE REVIEW

**Comment by soumya92** [comment]
This should fix #57, and maybe fix #69

**Comment by aarond10** [comment]
This is great! I've been meaning to do just this but haven't had the time. Thanks for the patch.
Please give me a few days to look over it.

File: src/logging.c [hunk 1]
```diff
@@ -66,6 +66,10 @@ void logging_cleanup() {
   logf = NULL;
 }
 
+int logging_debug_enabled() {
+  return LOG_DEBUG >= loglevel;
```
Comment (aarond10) [inline src/logging.c]:
```suggestion
  return loglevel <= LOG_DEBUG;
```

File: src/https_client.c [hunk 2]
```diff
@@ -316,14 +326,15 @@ void https_client_init(https_client_t *c, options_t *opt, struct ev_loop *loop)
 }
 
 void https_client_fetch(https_client_t *c, const char *url,
+                        const char* postdata, size_t postdata_len,
```
Comment (aarond10) [inline src/https_client.c]: FWIW, POST doesn't seem advised but I feel like caching (Read: MITM) should not be a thing over HTTPS so I'm totally OK with this. [GET support](https://tools.ietf.org/html/rfc8484#section-6) can be added by yourself or myself later.

[Seciton 5.1](https://tools.ietf.org/html/rfc8484#section-5.1)
```
   As a result, DoH servers need to carefully consider the HTTP caching
   metadata they send in response to GET requests (responses to POST
   requests are not cacheable unless specific response header fields are
   sent; this is not widely implemented and is not advised for DoH)
```

Comment (aarond10) [reply src/https_client.c]: Actually, now I re-read this, this paragraph is talking about server handling. Please disregard.

File: src/https_client.c [hunk 3]
```diff
@@ -176,6 +185,7 @@ static void https_fetch_ctx_cleanup(https_client_t *client,
       }
       curl_easy_cleanup(ctx->curl);
       cur->cb(cur->cb_data, cur->buf, cur->buflen);
+      free(cur->header_list);
```
Comment (aarond10) [inline src/https_client.c]:
```suggestion
      curl_slist_free_all(cur->header_list);
```

**Comment by soumya92** [comment]
Thanks for merging this in! I was out on a trip and didn't get a chance to address the comments, but I see you made the changes already :)

**Pull request merged**